### DISCUSSION TITLE
Documentation/examples for Yarn 2

### DISCUSSION THREAD

**Discussion by ineffyble** [batect/batect#1006] [General]
I haven't been able to find any info on how to set up caching right for Yarn 2 / Yarn Modern / Yarn Berry (not sure what the canonical name for it is).

There's two different approaches from what I can tell, as it has an optional global cache that's disabled by default.

I may be able to confirm what works and contribute it.

**Comment by charleskorn** [comment]
I don't have much experience with Yarn 2, so anything you can contribute would be great.

**Reply by ineffyble** [reply]
Turns out with Zero-Installs set up, there's no need for batect to care about cache at all.

**Reply by nth-commit** [reply]
Oooooh, that makes sense (from my extremely limited knowledge). Thanks.

**Comment by nth-commit** [comment]
FWIW (committing my WIP, in case I forget about this)

I didn't have much luck using legacy node resolution (node_modules et. al.) with batect and the node bundle - I was getting node_modules file lock errors on `yarn install`. A cursory go using zero-installs seems to get past this hurdle, at least (though it's going to be a learning curve for me).
